pandas/tests/io/parser/common/test_chunksize.py

```diff
@@ -190,8 +190,14 @@ def test_warn_if_chunks_have_mismatched_type(all_parsers, request):
     buf = StringIO(data)
 
     try:
-        with tm.assert_produces_warning(warning_type):
+        with tm.assert_produces_warning(warning_type) as record:
             df = parser.read_csv(buf)
+        if record:
+            expected = (
+                "Columns (0) have mixed types. Specify dtype option on import or "
+                "set low_memory=False."
+            )
+            assert str(record[0].message) == expected
```

Comment:
`tm.assert_produces_warning` now accepts a `match` argument

```suggestion
        msg = (
            "Columns (0) have mixed types. Specify dtype option on import or "
            "set low_memory=False."
        )
        with tm.assert_produces_warning(warning_type, match=re.escape(msg)):
            df = parser.read_csv(buf)
```